### PR TITLE
Add payload parameter support for `onConsumerRecord` method

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/2880
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    autoCommit: false
+};
+
+listener kafka:Listener kafkaListener = new (kafka:DEFAULT_URL, consumerConfigs);
+
+service kafka:Service on kafkaListener {
+    private final string var1 = "Kafka Service";
+    private final int var2 = 54;
+
+    remote function onConsumerRecord(kafka:ConsumerRecord[] records, kafka:Caller caller, @kafka:Payload PayloadConsumerRecord[] data) {
```
Comment (shafreenAnfar) [inline compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal:32]: Let's not use it `kafka:ConsumerRecord` as it is deprecated.

File: compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    string value;
+    int timestamp;
+    kafka:PartitionOffset offset;
+|};
+
+public type Person record {|
+    string name;
+    boolean isMarried;
+    int age;
+|};
```
Comment (shafreenAnfar) [inline compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal:156]: Also let's add scenario like the below.

```
remote function onConsumerRecord(@kafka:Payload kafka:MyRecord[] records)
```

Comment (dilanSachi) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal:178]: Updated in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/508/commits/c75fc2d0126153e4eb6265ec75ee9a04af23a17f

File: compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal (hunk 1 above)
Comment (dilanSachi) [reply compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal]: Updated in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/508/commits/c75fc2d0126153e4eb6265ec75ee9a04af23a17f

**Review by shafreenAnfar — APPROVED** [review]
LGTM!

**Comment by shafreenAnfar** [comment]
Just to confirm this does not introduce any breaking changes right ?

**Comment by dilanSachi** [comment]
> Just to confirm this does not introduce any breaking changes right ?

no

**Pull request merged**